commerce/domain/pricing/cart/PricingBasicCalculator/PricingCalculatorSample.cls

```diff
@@ -0,0 +1,264 @@
+/**
+* This Pricing calculator extension class makes a call to an external service or static to retrieve pricing
+* information for a cart item and saves it to a cart data transfer object (DTO).
+* For a pricing calculator extension to be processed by the checkout flow, you must implement the
+* CartExtension.PricingCartCalculator class.
+* This is just a sample for Pricing calculator, you are expected to refer this and write your own implementation.
+*/
+public class PricingCalculatorSample extends CartExtension.PricingCartCalculator {
+
+    private static Boolean useExternalService = false;
+
+    public virtual override void calculate(CartExtension.CartCalculateCalculatorRequest request) {
+        try {
+            CartExtension.Cart cart = request.getCart();
+            Iterator<CartExtension.CartItem> cartItemsIterator = clearErrorsAndGetCartItemsIterator(cart, request.getOptionalBuyerActionDetails());
+
+            // Get the SKUs from each cart item that needs price calculations
+            Map<String, CartExtension.CartItem> skuToCartItem = new Map<String, CartExtension.CartItem>();
```

Comment:
@pplat23 i am adding a filter here if (cartItem.getType() == CartExtension.SalesItemTypeEnum.PRODUCT), because thats how default pricing works. It doesnt calculate prices for shipping. So we shouldnt face any issues of multiple delivery group. Does that look good ? If no then how can this scenario be handled? I dont see solution for that in your doc yet.

Reply:
@swati681 specific problem with your code is that you are initializing a Map of SKU to CartItem in line 18 here. In line 21 you are updating it with an entry of SKU to CartItem. If there are multiple CartItems with the same SKU - duplicates will be overriden. In line 35 you are retrieving the values from the Map to update prices. As a result duplicate items for the same SKU will never be updated with a Price. Does this make sense?

I feel in order to remediate this situation you should use Set of SKUs and retrieve price per SKU. But then you have to iterate over ALL cartItems.

And could you please also add a test for this scenario?

Reply:
Got it, will add this.

Reply:
I have added changes for duplicate scenario and test.